### ISSUE TITLE
LR scheduler:  Error: 'Sine' object has no attribute '_step_count'

### ISSUE THREAD

**Issue by Mr-Ye-Cao** [bghira/SimpleTuner#1137]
Epoch 1/22, Steps:   0%|                                        | 29/30000 [01:08<17:47:11,  2.14s/it, lr=0, step_loss=0.245]2024-11-10 19:28:47,024 [ERROR] Failed to get the last learning rate from the scheduler. Error: 'Sine' object has no attribute '_step_count'
Epoch 1/22, Steps:   0%|                                        | 30/30000 [01:11<18:10:36,  2.18s/it, lr=0, step_loss=0.023]2024-11-10 19:28:49,022 [ERROR] Failed to get the last learning rate from the scheduler. Error: 'Sine' object has no attribute '_step_count'
Epoch 1/22, Steps:   0%|                                        | 31/30000 [01:13<17:42:51,  2.13s/it, lr=0, step_loss=0.118]2024-11-10 19:28:51,085 [ERROR] Failed to get the last learning rate from the scheduler. Error: 'Sine' object has no attribute '_step_count'
Epoch 1/22, Steps:   0%|                                       | 32/30000 [01:14<19:15:41,  2.31s/it, lr=0, step_loss=0.0314]^C

**Comment by Mr-Ye-Cao** [comment]
@bghira hi! i love ur product btw. 
i want to know if there's a branch i can use right now to avoid this issue ?

**Comment by bghira** [comment]
you can use a different LR scheduler, like 'constant'

**Comment by Mr-Ye-Cao** [comment]
yeah, i tried that before. but all differnt LR scheduler would give same error:
Epoch 1/22, Steps: 0%| | 29/30000 [01:08<17:47:11, 2.14s/it, lr=0, step_loss=0.245]2024-11-10 19:28:47,024 [ERROR] Failed to get the last learning rate from the scheduler. Error: 'Sine' object has no attribute '_step_count'

specifically, i tried to set: "LR_SCHEDULER=constant" in config.env

**Comment by bghira** [comment]
you should be using config.json

**Comment by Mr-Ye-Cao** [comment]
thanks a lot! i tried config.json this time. the default (which i guess is polynomial) still same error happens. but constant works this time!

**Comment by bghira** [comment]
i fixed sine, but polynomial already worked when i tested
